### DISCUSSION TITLE
SQL Editor region collapse like in JS with VS Code

### DISCUSSION THREAD

**Discussion by jonashaus** [supabase/supabase#33570] [Feature Requests]
In most IDE's, you can create collapsible "regions" by wrapping Javascript code with region comments:

```
//#region MyRegion

[some JS code]

//#endregion
```

The IDE will show this as a collapsible code block - i use this to improve code readability. Since the Supabase SQL Editor "IDE" already supports collapsing e.g. "coalesce" statements, I thought this would also be cool for regions. Of course, in SQL, the syntax would be:

```
--#region MyRegion

[some SQL]

--#endregion
```

I personally would love to see this feature and hope to see some general community interest in this 😊

**Comment by zjy4fun** [comment]
Great idea!
But I think the SQL Editor of Supabase is just a simple editor and probably won't introduce this feature.

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.
